api/integrations/github/github.py

```diff
@@ -81,22 +89,37 @@ def generate_body_comment(
     last_updated_string = LAST_UPDATED_FEATURE_TEXT % (
         datetime.datetime.now().strftime("%dth %b %Y %I:%M%p")
     )
-    updated_text = UPDATED_FEATURE_TEXT % (name)
 
-    result = updated_text if is_update else LINK_FEATURE_TEXT % (name)
+    result = UPDATED_FEATURE_TEXT % (name) if is_update else LINK_FEATURE_TITLE % (name)
+    last_segment_name = ""
+    if len(feature_states) > 0 and not feature_states[0].get("segment_name"):
+        result += FEATURE_TABLE_HEADER
 
-    # if feature_states is None:
     for v in feature_states:
         feature_value = v.get("feature_state_value")
         feature_value_type = v.get("feature_state_value_type")
+        tab = "segment-overrides" if v.get("segment_name") is not None else "value"
+        environment_link_url = FEATURE_ENVIRONMENT_URL % (
+            get_current_site_url(),
+            project_id,
+            v.get("environment_api_key"),
+            feature_id,
+            tab,
+        )
+        if v.get("segment_name") is not None and v["segment_name"] != last_segment_name:
+            result += "\n" + LINK_SEGMENT_TITLE % (v["segment_name"])
+            last_segment_name = v["segment_name"]
+            result += FEATURE_TABLE_HEADER
+        table_row = FEATURE_TABLE_ROW % (
+            v["environment_name"],
+            environment_link_url,
+            "✅ Enabled" if v["feature_value"] else "❌ Disabled",
```

Comment:
I don't think the logic here is correct. Disabled feature states can have values.

I'm not entirely sure where the feature state data comes from, but, assuming it has the "enabled" key, the code should be:

```suggestion
            "✅ Enabled" if v["enabled"] else "❌ Disabled",
```

Reply:
The logic is correct, but the variable name is confusing. Variable renamed